### ISSUE TITLE
sharing system.img dump with good stuffs

### ISSUE THREAD

**Issue by ziegfeld** [HappyZ/dpt-tools#37]
**How to Use**
This system.img (or if you prefer, download smaller-sized systemimg.tgz and unarchive) can be applied by dpt-tools.py `restore-systemimg` in diagnosis mode, after you applied all the HappyZ's dpt-tools hacks.

Just putting it here as a milestone or snapshot of current state, so that more people can involve in and make improvements.
Feel free to test it. It would not brick your DPT-RP1 anyway :)
'
[GoogleDrive Folder](https://drive.google.com/drive/folders/1tVKM2A6ST5-_jgZ4izdXmkh8mPdYFYsd?usp=sharing)

You may encounter AppLauncher/Settings not working problem, please try ["`step 5` in `Launcher app` section under `0x4 Other Tips`"](https://github.com/HappyZ/dpt-tools/blob/master/README.md#0x4-other-tips) and also make sure to grant su access in the SuperSu prompt. See posts below for details or leave your question.

For CHINESE users: after boot normally, press menu button, choose "Eink settings" then "Launguage", then you can select "中文”. Go back once (using Button Savior) and choose "setup IME" you can enable Google Pinyin (it's minimal, e-ink optimized and power-saving) and disable all the other 3 IMEs.

**How did we make this**
It is based on Sony's official firware 1.4.0.1, with some files copied from the system.img of kdroid.club dpt1-user 1211 update: `app/{Eink.apk,PinyinIME.apk,applauncher.apk,buttonsavior.apk,filemanager.apk,Superuser.apk,Bluetooth/} etc/dp_extensions/ bin/adb vendor/recv.bin xbin build.prop lib/{libjnidemo.so,libwebviewchromium.so,libskia.so}`

**What do you get?**
Kudos for @HappyZ @p4s2wd who helped getting to here step-by-step; I only follow orders :D . They know what's working in this and where to go next:)
1. Bluetooth keyboard works!
2. pressing menu button you will get more icons for other apps like a file manager, the web browser, an "E-ink Apps" launcher appp (and Kindle works if you install it)
3. SuperSu for superuser permission management
4. Button Savior Pro

**what's not working (and need your help to develop)**
1. taobao "settings app" - most settings (USB storage mode switch, screen dpi change, contrast/brightness bar)
2. Bluetooth receiving file -- no way to choose "accept this file" in RP1. (But HID keyboard works!)

**Misc.**
I can share a repo of working e-inkified or minimalist apks for this, including e-inkified RE explorer, reading apps like ALReader, e-ink App Market, Greenify and IceBox for power saving, screen rotation setter, ssh server and X-plor, and some Xposed modules verifiedly working with this later. Or you can explore hi-pda.com e-ink bbs (in Chinese).

Edit: tips on Bluetooth Keyboard.
Please do use Google Pinyin Keyboard -- I found the other 3 either cashing or not able to use for Chinese Pinyin Input.

1. Shift+Space will switch English and Chinese language mode.
2. in Chinese Pinyin mode, you can see the candidate Chinese characters in the bottom of DPT display; and the English letters you already input are really small above that.
    Use 1,2,..,9 to choose the character you want;
    or use Left/Right Arrow to navigate, Up/Down Arrow for next/last page of candidates (instead of PageUp/PageDown); 
    and Space to choose the highlighted character.
3. "Windows" or "Apple Command" key will bring up AppLauncher, i.e. the pop-up menu the same as when the physical menu button is pressed.

--- you are welcome to share your findings and thoughts!
See also Work In Progress: #32 #36 #28 #38

**Comment by nyanadassana** [comment]
I would like to test, but can you explain this"by dd in diagnosis mode"? I can't find dd in the diagnosis mode. Thanks.

**Comment by HappyZ** [comment]
@nyanadassana you should be able to run `restore-systemimg` in the tool (in diagnosis mode) and follow the instruction to flash. When it asks if your system.img is a sparse image, say `no`. 

ATTENTION: if you come from somewhere else, the official `system.img` is actually a sparse image, do say `yes` if you use the official system image! Do NOT answer `no` here. If you did, do NOT turn off the device. Reformat your system partition and restore the system image.

**Comment by nyanadassana** [comment]
Success! but Applaucher stop working...

**Comment by HappyZ** [comment]
> Success! but Applaucher stop working...

Try step 5 in `Launcher app` section under `0x4 Other Tips`

**Comment by nyanadassana** [comment]
tried,  Permission denied

**Comment by nyanadassana** [comment]
it's seems SuperSU controls the root permission now?

**Comment by nyanadassana** [comment]
After I give permission to shell in SuperSU, works now!

**Comment by nyanadassana** [comment]
![dp_20181215_1018](https://user-images.githubusercontent.com/28098044/50038763-19fff080-0053-11e9-8f51-149fc2b33da3.png)

**Comment by nyanadassana** [comment]
is there anything I can help with testing?

**Comment by sekkit** [comment]
hi is any thirdparty bluetooth app works for file transfer？
thx but why port libjnidemo.so,libwebviewchromium.so,libskia.so, do they have any influence to functionality?

**Comment by ziegfeld** [comment]
> hi is any thirdparty bluetooth app works for file transfer？
> thx but why port libjnidemo.so,libwebviewchromium.so,libskia.so, do they have any influence to functionality?

That would be a question for the guru @p4s2wd

**Comment by p4s2wd** [comment]
> > hi is any thirdparty bluetooth app works for file transfer？
> > thx but why port libjnidemo.so,libwebviewchromium.so,libskia.so, do they have any influence to functionality?
> 
> That would be a question for the guru @p4s2wd

These libs were added by Taobao's seller.

**Comment by sh-miller** [comment]
@ziegfeld    nice work! 
after i flash the system.img , the  applauncher  stoped when DPT-RP1 started.
following log is from logcat  : 

W/BroadcastQueue( 2136): Unable to launch app com.sony.apps.applauncher/10014 for broadcast Intent { act=com.sony.infras.DPExtensions.MenuKeyHandler.MENU_KEY_PUSHED flg=0x10 }: process is bad
D/PowerManagerService( 2136): [INTENT] ACTION_SCREEN_ON_FROM_SNOOZE

**Comment by nyanadassana** [comment]
@sh-miller do what happyZ mentioned above.

**Comment by sh-miller** [comment]
> @sh-miller do what happyZ mentioned above.

thanks for you help,   i use  adb shell login DPT-RP1,  mount system, 
use 'su' switch to root user,   and  grant all permission to  /etc/dp_extensions folder,  
 chmod -R  777   /etc/dp_extensions,    how do you solve the problem ?

**Comment by nyanadassana** [comment]
you should do this:
Remove the database (cache) from the Extension Manager and allow it to rebuid the database:
cd /data/system
mv ExtMgr.db ExtMgr.db_bak
mv ExtMgr.db-journal ExtMgr.db-journal_bak
reboot

**Comment by ziegfeld** [comment]
> @ziegfeld nice work!
> after i flash the system.img , the applauncher stoped when DPT-RP1 started.
> following log is from logcat :
> 
> W/BroadcastQueue( 2136): Unable to launch app com.sony.apps.applauncher/10014 for broadcast Intent { act=com.sony.infras.DPExtensions.MenuKeyHandler.MENU_KEY_PUSHED flg=0x10 }: process is bad
> D/PowerManagerService( 2136): [INTENT] ACTION_SCREEN_ON_FROM_SNOOZE

What @nyanadassana said is the right way.
I GUESS I did not meet this because that I did factory_reset FIRST BEFORE flashing this system.img.
In diagnosis mode, do a "factory_reset" to clear all your data ( just like android phone factory reset, clearing all your downloaded apps, prefs and files):
    /usr/local/bin/factory_reset.sh

**Comment by sh-miller** [comment]
> you should do this:
> Remove the database (cache) from the Extension Manager and allow it to rebuid the database:
> cd /data/system
> mv ExtMgr.db ExtMgr.db_bak
> mv ExtMgr.db-journal ExtMgr.db-journal_bak
> reboot

@nyanadassana  thanks,    it works!

**Comment by sh-miller** [comment]
> > @ziegfeld nice work!
> > after i flash the system.img , the applauncher stoped when DPT-RP1 started.
> > following log is from logcat :
> > W/BroadcastQueue( 2136): Unable to launch app com.sony.apps.applauncher/10014 for broadcast Intent { act=com.sony.infras.DPExtensions.MenuKeyHandler.MENU_KEY_PUSHED flg=0x10 }: process is bad
> > D/PowerManagerService( 2136): [INTENT] ACTION_SCREEN_ON_FROM_SNOOZE
> 
> What @nyanadassana said is the right way.
> I GUESS I did not meet this because that I did factory_reset FIRST BEFORE flashing this system.img.
> In diagnosis mode, do a "factory_reset" to clear all your data ( just like android phone factory reset, clearing all your downloaded apps, prefs and files):
> /usr/local/bin/factory_reset.sh

@ziegfeld   thanks,   now it works,   i had installed some app in applauncher     before I flash the system.img that your shared.   I  GUESS flash system.img will reset the app setting or other data.  so not  do this:
mv ExtMgr.db ExtMgr.db_bak
mv ExtMgr.db-journal ExtMgr.db-journal_bak
yea, It's my fault。
here is  a episode.   in diagnosis mode, exec cmd "restore-systemimg", copy the system.img to the   mass_storage in MacOX( not pc),  then delete the system.img  in MacOX  after, the  disk space is not release,  
if    “ ls   /mnt/sdtmp ” ， the folder is emtpy,   but the deleted system.img is in  folder  "/mnt/sdtmp/.Trashes/501# ",  rm it ,  then the disk space will  real release.

**Comment by nyanadassana** [comment]
There is a new issue shows: it appears Bluetooth share has stoped. Anyone help to fix this?

**Comment by iuyo5678** [comment]
thanks ,My device DPT-CP1 has use this firmware but I found a problem, the device can't connect the digital paper app, I Guess I use the mass_storage  setting  because when I connect the device to mac  ,I found  the mac mount a storage named Unitted in desktop. how can I  close the setting. or How can i use the digital pager app  ?

**Comment by HappyZ** [comment]
@iuyo5678 via wifi, or bluetooth

**Comment by iuyo5678** [comment]
> @iuyo5678 via wifi, or bluetooth

usb cable , and I found  use the NFC my phone can connect to the device. but use usb cable the mac can't

**Comment by iuyo5678** [comment]
@HappyZ  thanks I fixed this problem ，in the diag mode   /usr/local/bin/factory_reset.sh   and restore_boot.img  and restore system.img

**Comment by tartakis** [comment]
> I can share a repo of working e-inkified or minimalist apks for this...

@ziegfeld: Could you please do so? 
And have you (or somebody else for that matter) been succesful at installing VNC viewer on the DPT-RP1? I want to use DPT as a monitor for my mac and so far VNC crashes after installing it.

**Comment by ziegfeld** [comment]
@tartakis you can try teamviewer and iDisplay here
http://182.254.232.41/apptest/SONY/

**Comment by sisyphose** [comment]
CP1 use this img Bluetooth will crash and always show the pop，I guess this is because the img you posted is rp1‘s, so can you provide the cp1 file?  thank you.

**Comment by tartakis** [comment]
> @tartakis you can try teamviewer and iDisplay here
http://182.254.232.41/apptest/SONY/

For other people who may want to try it, the package name for TeamViewer is `com.teamviewer.teamviewer.market.mobile` and the Android app intent name is `com.teamviewer.teamviewer.market.mobile/com.teamviewer.remotecontrollib.activity.MainActivity.`

You can launch it from adb shell using `am start -n com.teamviewer.teamviewer.market.mobile/com.teamviewer.remotecontrollib.activity.MainActivity`

**Comment by HappyZ** [comment]
> CP1 use this img Bluetooth will crash and always show the pop，I guess this is because the img you posted is rp1‘s, so can you provide the cp1 file? thank you.


I'd love to see a crash report from logcat to see what the real problem is.

**Comment by sisyphose** [comment]
> > CP1 use this img Bluetooth will crash and always show the pop，I guess this is because the img you posted is rp1‘s, so can you provide the cp1 file? thank you.
> 
> I'd love to see a crash report from logcat to see what the real problem is.

I'm sorry ， I don't know how to find the crash report. Don't flash the taobao's img would not have the problem.
I have cracked three cp1s and will crash Bluetooth. I am sure that this is a common problem, not a case.

**Comment by sisyphose** [comment]
@ziegfeld hello，Are you interested in making a new sisytem.img? If so, please contact me. Sisyphose@gmail.com

**Comment by ziegfeld** [comment]
> @ziegfeld hello，Are you interested in making a new sisytem.img? If so, please contact me. [Sisyphose@gmail.com](mailto:Sisyphose@gmail.com)

I am afraid not really, friend.. Been busy recently.
You may try to do it yourself with the instructions and ask questions here, if any.

**Comment by DeeDive** [comment]
@ziegfeld  What's the IME in your system.img? It's fantastic!

**Comment by ziegfeld** [comment]
@infinitestudy not my credit. Kudos to hipda eink pals. Well it’s part of the taobao image and a prebuilt modified google pinyin. Super light I know.

**Comment by wangjazz** [comment]
> **How to Use**
> This system.img (or if you prefer, download smaller-sized systemimg.tgz and unarchive) can be applied by dpt-tools.py `restore-systemimg` in diagnosis mode, after you applied all the HappyZ's dpt-tools hacks.
> 
> Just putting it here as a milestone or snapshot of current state, so that more people can involve in and make improvements.
> Feel free to test it. It would not brick your DPT-RP1 anyway :)
> '
> [GoogleDrive Folder](https://drive.google.com/drive/folders/1tVKM2A6ST5-_jgZ4izdXmkh8mPdYFYsd?usp=sharing)
> 
> You may encounter AppLauncher/Settings not working problem, please try ["`step 5` in `Launcher app` section under `0x4 Other Tips`"](https://github.com/HappyZ/dpt-tools/blob/master/README.md#0x4-other-tips) and also make sure to grant su access in the SuperSu prompt. See posts below for details or leave your question.
> 
> For CHINESE users: after boot normally, press menu button, choose "Eink settings" then "Launguage", then you can select "中文”. Go back once (using Button Savior) and choose "setup IME" you can enable Google Pinyin (it's minimal, e-ink optimized and power-saving) and disable all the other 3 IMEs.
> 
> **How did we make this**
> It is based on Sony's official firware 1.4.0.1, with some files copied from the system.img of kdroid.club dpt1-user 1211 update: `app/{Eink.apk,PinyinIME.apk,applauncher.apk,buttonsavior.apk,filemanager.apk,Superuser.apk,Bluetooth/} etc/dp_extensions/ bin/adb vendor/recv.bin xbin build.prop lib/{libjnidemo.so,libwebviewchromium.so,libskia.so}`
> 
> **What do you get?**
> Kudos for @HappyZ @p4s2wd who helped getting to here step-by-step; I only follow orders :D . They know what's working in this and where to go next:)
> 
> 1. Bluetooth keyboard works!
> 2. pressing menu button you will get more icons for other apps like a file manager, the web browser, an "E-ink Apps" launcher appp (and Kindle works if you install it)
> 3. SuperSu for superuser permission management
> 4. Button Savior Pro
> 
> **what's not working (and need your help to develop)**
> 
> 1. taobao "settings app" - most settings (USB storage mode switch, screen dpi change, contrast/brightness bar)
> 2. Bluetooth receiving file -- no way to choose "accept this file" in RP1. (But HID keyboard works!)
> 
> **Misc.**
> I can share a repo of working e-inkified or minimalist apks for this, including e-inkified RE explorer, reading apps like ALReader, e-ink App Market, Greenify and IceBox for power saving, screen rotation setter, ssh server and X-plor, and some Xposed modules verifiedly working with this later. Or you can explore hi-pda.com e-ink bbs (in Chinese).
> 
> Edit: tips on Bluetooth Keyboard.
> Please do use Google Pinyin Keyboard -- I found the other 3 either cashing or not able to use for Chinese Pinyin Input.
> 
> 1. Shift+Space will switch English and Chinese language mode.
> 2. in Chinese Pinyin mode, you can see the candidate Chinese characters in the bottom of DPT display; and the English letters you already input are really small above that.
>    Use 1,2,..,9 to choose the character you want;
>    or use Left/Right Arrow to navigate, Up/Down Arrow for next/last page of candidates (instead of PageUp/PageDown);
>    and Space to choose the highlighted character.
> 3. "Windows" or "Apple Command" key will bring up AppLauncher, i.e. the pop-up menu the same as when the physical menu button is pressed.
> 
> --- you are welcome to share your findings and thoughts!
> See also Work In Progress: #32 #36 #28 #38

Is this dump system image work for CP1 as well? or alternative method to obtain the useful apks and install one-by-one manually？

**Comment by jacky1019** [comment]
Thanks for this great stuff! I was able to install it, but Kindle doesn't seem to work. It's shown in the app launcher, but nothing happens after I click it. 

If I copy the Kindle apk to the storage, and use File Manager to install it, and click "Open" after it's successfully installed, it can be launched and works perfectly. However, if I switch to other app, I can never use the Kindle icon in the app launch to switch back to it. I noticed the Kindle_extention.xml actually uses the UpgradePage as the intent. Is this expected? How can I figure out the correct intent name?
